wagtail/contrib/redirects/views.py

```diff
@@ -36,66 +37,88 @@
 )
 from wagtail.log_actions import log
 
-permission_checker = PermissionPolicyChecker(permission_policy)
 
+class RedirectPageColumn(Column):
+    cell_template_name = "wagtailredirects/redirect_page.html"
 
-@permission_checker.require_any("add", "change", "delete")
-@vary_on_headers("X-Requested-With")
-def index(request):
-    query_string = request.GET.get("q", "")
-    ordering = request.GET.get("ordering", "old_path")
+    def get_cell_context_data(self, instance, parent_context):
+        context = super().get_cell_context_data(instance, parent_context)
+        context["redirects"] = instance
+        return context
 
-    redirects = models.Redirect.objects.prefetch_related("redirect_page", "site")
 
-    # Search
-    if query_string:
-        redirects = redirects.filter(
-            Q(old_path__icontains=query_string)
-            | Q(redirect_page__url_path__icontains=query_string)
-            | Q(redirect_link__icontains=query_string)
-        )
+permission_checker = PermissionPolicyChecker(permission_policy)
 
-    # Ordering (A bit useless at the moment as only 'old_path' is allowed)
-    if ordering not in ["old_path"]:
-        ordering = "old_path"
 
-    redirects = redirects.order_by(ordering)
+class Index(IndexView):
+    template_name = "wagtailredirects/index.html"
+    results_template_name = "wagtailredirects/results.html"
```

Comment:
Let's rename the template while we're here

```suggestion
    results_template_name = "wagtailredirects/index_results.html"
```